test/rulesets/Base/base.jl

```diff
@@ -87,11 +87,13 @@
 
             x isa Real && test_scalar(cbrt, x)
             if (x isa Real && x >= 0) || x isa Complex
-                test_scalar(sqrt, x)
-                test_scalar(log, x)
-                test_scalar(log2, x)
-                test_scalar(log10, x)
-                test_scalar(log1p, x)
+                # this check is needed because the imaginary part jumps from pi to -pi
+                nr = imag(x) != 0 && real(x) < 0
+                test_scalar(sqrt, x, test_wirtinger=nr)
+                test_scalar(log, x, test_wirtinger=nr)
+                test_scalar(log2, x, test_wirtinger=nr)
+                test_scalar(log10, x, test_wirtinger=nr)
+                test_scalar(log1p, x, test_wirtinger=nr)
```

Comment:
kwargs should be seperated from non-kwargs by a `;`

Also can we rename `nr` to something clearer.
like `should_test_wirtinger`?
Or maybe we can move it into the definition of `test_scalar` ?

Reply:
I can definitely rename it. The problem with moving this into the definition of `test_scalar` is that it only applies to these particular functions since these have discontinuities between `-10 + im*eps()` and `-10 - im*eps()`.